### PR TITLE
Updated is_valid_block() and test0[1,2,3,4]_is_valid_block() for clarity

### PR DESCRIPTION
**Pull request by jldistefano**

### CODE REVIEW

File: data_structures/block.py [hunk 1]
```diff
@@ -390,3 +390,26 @@ def parse_block(block_header):
...
+    """
+    block_info = parse_block(block)
+    prev_block_info = parse_block(prev_block)
+    # Ensures that time timestamp of block is greater than the timestamp of prev_block
+    if (bytes_to_int(block_info["timestamp"]) <= bytes_to_int(prev_block_info["timestamp"])):
+        return False
+    # Ensures that the prev_hash element of block matches the hash of prev_block
+    if (block_info["prev_hash"] != hash_SHA(prev_block)):
+        return False
+    # Ensures that the block was mined correctly, and the target element of block is less than the hash of block
```
Comment (kstola2) [inline data_structures/block.py]: This comment is incorrect, the logic is that the block hash _should_ be less than the target.

File: data_structures/block.test.py [hunk 2]
```diff
@@ -266,8 +266,78 @@ def test_parse_block(self):
        self.assertEqual(slice_nonce(header), parsed_block["nonce"])
        self.assertEqual(slice_timestamp(header), parsed_block["timestamp"])
        self.assertEqual(hash_SHA(header), parsed_block["block_hash"])
-
-
+      
+    # This test ensures that a block with a timestamp less than the timestamp of its previous block, will not be added to the blockchain
+    def test_01_is_valid_block(self):
+        target = 10**72     # Target for which all block hashes must be under
+        data = hash_SHA("1".encode())   # valid block data to be used for block constructoin
```
Comment (kstola2) [inline data_structures/block.test.py]: Spell check here

File: data_structures/block.test.py [hunk 3]
```diff
@@ -266,8 +266,78 @@ def test_parse_block(self):
...
+        prev_hash = hash_SHA("0".encode())  # This previous hash is invalid, though the timestamp of the block will be checked first
+        #Creates an invalid block, this is invalid due to having a timestamp less than the candidate block
+        invalid_candidate_block = mine(prev_hash, hash_SHA("INVALID".encode()), target)
+
+        #Creates a valid previous block
+        prev_block = mine(hash_SHA("0".encode()), data, target)
+        prev_hash = hash_SHA(prev_block)
```
Comment (kstola2) [inline data_structures/block.test.py]: What is the purpose of setting the prev_hash here?

File: data_structures/block.test.py [hunk 4]
```diff
@@ -266,8 +266,78 @@ def test_parse_block(self):
...
+        target = 10**72     # Target for which all block hashes must be under
+        data = hash_SHA("1".encode())   # valid block data to be used for block constructoin
+
+        #This previous hash would be invalid if used to create a new block, as it does not match hash_SHA(prev_block)
+        invalid_prev_hash = hash_SHA("BEEPBEEPLETTUCE".encode())
+
+        #Creates a valid previous block
+        prev_block = mine(hash_SHA("0".encode()), data, target)
+
+        time.sleep(2) # Ensures prev_block and candidate block will not have identical timestamps
```
Comment (kstola2) [inline data_structures/block.test.py]: I think 1 second is enough for this test.

File: data_structures/block.test.py [hunk 5]
```diff
@@ -266,8 +266,78 @@ def test_parse_block(self):
...
+    def test_03_is_valid_block(self):
+        previous_target = 10**75     # Target for which all block hashes must be under
+        candidate_target = 1        # A low target like this will create a block with an invalid hash
+        data = hash_SHA("1".encode())   # valid block data to be used for block constructoin
+
+        #Creates a valid previous block
+        prev_block = mine(hash_SHA("0".encode()), data, previous_target)
+        prev_hash = hash_SHA(prev_block)
+
+        time.sleep(2) # Ensures prev_block and candidate block will not have identical timestamps
```
Comment (kstola2) [inline data_structures/block.test.py]: 1 sec

File: data_structures/block.test.py [hunk 6]
```diff
@@ -266,8 +266,78 @@ def test_parse_block(self):
...
+
+    # This tests shows that a valid block, a block that meets none of the above three failure states, will be added to the blockchain
+    def test_04_is_valid_block(self):
+        target = 10**72     # Target for which all block hashes must be under
+        data = hash_SHA("1".encode())   # valid block data to be used for block constructoin
+        #Creates a valid previous block
+        prev_block = mine(hash_SHA("0".encode()), data, target)
+        prev_hash = hash_SHA(prev_block)
+
+        time.sleep(2)   # Ensures prev_block and candidate block will not have identical timestamps
```
Comment (kstola2) [inline data_structures/block.test.py]: 1 sec

**Comment by jldistefano** [comment]
Requested Changes Made.

**Comment by kstola2** [comment]
Changes approved. Feel free to delete your branch.

**Pull request merged**